com.unity.render-pipelines.high-definition/Runtime/Lighting/LightCookieManager.cs

```diff
@@ -268,17 +412,135 @@ public void ReserveSpace(Texture cookie)
+                                        invSize, invSize, 0.5f*invSize, 0.5f*invSize);
+
+            const int groupSizeX = 8;
+            const int groupSizeY = 8;
+            int threadGroupX = (projectionSize + (groupSizeX - 1))/groupSizeX;
+            int threadGroupY = (projectionSize + (groupSizeY - 1))/groupSizeY;
+
+            cmd.DispatchCompute(m_ProjectCubeTo2D, usedKernel, threadGroupX, threadGroupY, 1);
+
+            return projected;
+        }
+
+        public Vector4 FetchCubeCookie(CommandBuffer cmd, Texture cookie)
+        {
+            Debug.Assert(cookie != null);
+            Debug.Assert(cookie.dimension == TextureDimension.Cube);
+
+            int projectionSize = 2*(int)Mathf.Max((float)m_CookieCubeResolution, (float)cookie.width);
+            if (projectionSize < k_MinCookieSize)
+                return Vector4.zero;
+
+            if (!m_CookieAtlas.IsCached(out var scaleBias, cookie) && !m_NoMoreSpace)
+                Debug.LogError($"Cube cookie texture {cookie} can't be fetched without having reserved. You can try to increase the cookie atlas resolution in the HDRP settings.");
+
+            if (m_CookieAtlas.NeedsUpdate(cookie, true))
+            {
+                Vector4 sourceScaleOffset = new Vector4(projectionSize/(float)atlasTexture.rt.width, projectionSize/(float)atlasTexture.rt.height, 0, 0);
+
+                RTHandle projectedCookie = ProjectCubeTo2D(cmd, cookie, projectionSize);
+
+                // Generate the mips
+                Texture filteredProjected = FilterAreaLightTexture(cmd, projectedCookie);
+                m_CookieAtlas.BlitOctahedralTexture(cmd, scaleBias, filteredProjected, sourceScaleOffset, blitMips: true, overrideInstanceID: m_CookieAtlas.GetTextureID(cookie));
+
+                RTHandlesDeleter.ScheduleRelease(projectedCookie);
+            }
+
+            return scaleBias;
+        }
+
+        public Vector4 FetchCubeCookie(CommandBuffer cmd, Texture cookie, Texture ies)
+        {
+            Debug.Assert(cookie != null);
+            Debug.Assert(ies != null);
+            Debug.Assert(cookie.dimension == TextureDimension.Cube);
+            Debug.Assert(ies.dimension == TextureDimension.Cube);
+
+            int projectionSize = 2*(int)Mathf.Max((float)m_CookieCubeResolution, (float)cookie.width);
+            if (projectionSize < k_MinCookieSize)
+                return Vector4.zero;
+
+            if (!m_CookieAtlas.IsCached(out var scaleBias, cookie, ies) && !m_NoMoreSpace)
+                Debug.LogError($"Cube cookie texture {cookie} can't be fetched without having reserved. You can try to increase the cookie atlas resolution in the HDRP settings.");
+
+            if (m_CookieAtlas.NeedsUpdate(cookie, ies, true))
+            {
+                Vector4 sourceScaleOffset = new Vector4(projectionSize/(float)atlasTexture.rt.width, projectionSize/(float)atlasTexture.rt.height, 0, 0);
```

Comment:
something I am not sure to understand. When the light only have a cookie. It seems that we always perform the  
PUArithmetic.ComputeOperation(projectedCookie, projectedIES, cmd, GPUArithmetic.Operation.Mult); operation right?
This shoul dbe avoided, if the operation is not needed we should do it (in particular if we have an animated cookie without IES for example. And it doesn't require extra allocation either.

Reply:
ok, seems like this is done later and external to this function, right?

Reply:
It's needed to compute the Mips too. Could find a way to add a parameter like "IsMipsAlreadyComputed" on the IES or Cookie provided.